### ISSUE TITLE
Can't import flow types with sourceType script

### ISSUE THREAD

**Issue by Hilzu** [babel/babel-eslint#474]
I'm writing an app with CommonJS modules and Flowtype. When setting sourceType to `script` it becomes impossible to use the Flow `import type` syntax. As far as I can tell Flow only has ES2015 style syntax for type imports.

Here's my eslint configuration:
```json
{
  "parser": "babel-eslint",
  "parserOptions": {
    "sourceType": "script"
  },
  "plugins": [
    "flowtype",
  ],
  "extends": [
    "eslint:recommended",
    "plugin:flowtype/recommended"
  ]
}
```

Code like this triggers parse errors:
```js
const { promisify } = require("./promise")
import type { Context } from "koa" 
```

Example error:
```
  15:1  error  Parsing error: 'import' and 'export' may appear only with 'sourceType: module'

  13 | const { promisify } = require("./promise")
  14 |
> 15 | import type { Context } from "koa"
     | ^
```

Is there a way to use CommonJS modules and the `import type` syntax together with `babel-eslint` ?

**Comment by kaicataldo** [comment]
Thank you for the issue. Now that [`@babel/eslint-parser` has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).
